src/createStore.js

```diff
@@ -238,15 +140,12 @@ export default function createStore(reducer, initialState, enhancer) {
     }
   }
 
-  // When a store is created, an "INIT" action is dispatched so that every
-  // reducer returns their initial state. This effectively populates
-  // the initial state tree.
-  dispatch({ type: ActionTypes.INIT })
+  replaceReducer(reducer)
 
   return {
     dispatch,
```

Comment:
The problem I tried to solve is that anything other than `{ getState,  dispatch }` is hard to compose, and enhancers have no knowledge of the past enhancers. If any enhancer could add a method like `subscribe()`, it would be a pain to later batch subscriptions in any later enhancer (this is pretty much why https://github.com/tappleby/redux-batched-subscribe is currently using a less performant subscription mechanism which was copy pasted from earlier versions of Redux). If any enhancer could add `[$$observable]`, then any further enhancer changing `getState` would cause it to report incorrect values because the inner enhancer uses the inner `getState`.

Sealing off limits the domain of enhancers to:

* Hijacking `reducer`
* Hijacking `initialState`
* Hijacking `dispatch`
* Hijacking `getState`
* Hijacking `onChange`

All these things are guaranteed to be orthogonal: it is always possible to write an enhancer that changes some of them and it is guaranteed that this won’t trip up enhancers before or after it.

However as soon as we let enhancers define non-orthogonal methods (such as `[$$observable]` which depends on `subscribe` and `getState`), this introduces [subtle failures](https://github.com/gaearon/redux-devtools/pull/275). If a default enhancer included `subscribe`, a thing that wants to *delay* `subscribe` would have to reimplement its logic.

I feel good that all existing use cases seem covered by `(reducer, initialState, onChange) => { dispatch, getState }`. I would be wary of making the system more expressive because this caused issues in the past. Most enhancers so far had to *work around* this freedom rather than enjoy it. Finally, for what it’s worth, it is always possible to *really* wrap `createStore` if you *really* need it. I’m just not convinced this is any close to being a common case for the enhancers I have seen.